test/test_pycparserext.py

```diff
@@ -374,6 +374,30 @@ def test_double_pointer():
     assert gen.visit(ast).find("func_with_p2pp(const char *, Error **)") != -1
 
 
+def test_designated_initializers():
+    src = """
+    int a[6] = { [4] = 29, [2] = 15 };
+
+    int widths[] = { [0 ... 9] = 1, [10 ... 99] = 2, [100] = 3 };
+
+    int v1 = 5, v2 = 6;
+    int b[] = { [1] = v1, v2, [4] = v4 };
+
+    struct foo { int x; int y; }; 
+    struct foo bar[10] = { [1].y = 5, [2].x = 1, [0].x = 3 };
+    
+    char char_map[256] = { [0 ... 255] = '?', ['0' ... '9'] = 'X' }; 
+    """
+    import pycparserext.ext_c_parser as ext_c_parser
+    import pycparserext.ext_c_generator as ext_c_generator
+
+    parser = ext_c_parser.GnuCParser()
+    ast = parser.parse(src)
+    ast.show()
+    gen = ext_c_generator.GnuCGenerator()
+    print(gen.visit(ast))
```

Comment:
I know none of the other tests do this, but it would be useful to test this as an identical parser/generator round-trip. If you're game, OK, otherwise, also fine.

Reply:
Sure! I'll take a look at getting this to round-trip.